asm/single_precision

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+echo Generating dsdot.go
+echo '// Generated code do not edit. Run `go generate`.\n' > dsdot.go
+cat ddot.go \
```

Comment:
If you end each line with a `|`, you don't need the backslashes.

I think if we're doing lots of code generations like this, we should move to using Go instead of sh. I've done something like this before (see https://github.com/fhs/go-netcdf/blob/master/netcdf/generate.go). It's fine for now since it's just few functions.